app/views/submissions/show.html.erb

```diff
@@ -31,8 +31,14 @@
     <% end %>
   <% end %>
 </div>
-<%= form_for(new_comment, url: submission_create_comment_path(post.id), method: :post) do |f| %>
-  <%= f.text_area :text, cols: 60 %>
+<%= form_for(new_comment, url: submissions_path, method: :post) do |comment| %>
+  <%= comment.text_area :text, cols: 60 %>
   <br />
-  <%= f.submit "add comment" %>
+  <%= comment.hidden_field :post_id, :value => parent_id %>
+  <% flash.each do |type, msg| %>
+    <% if type.to_s == "inline" %>
```

Comment:
hmmm there must be a better way to handle re-rendering someone's comment -- especially if they've typed a really long comment in and some other error happened and they lost their post.

Reply:
This one was tricky.... It may be another reason why splitting the model could help